src/App.tsx

```diff
@@ -1,27 +1,52 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.scss';
 import { GoodsList } from './GoodsList';
+import { Good } from './types/Good';
+import 'bulma';
 
-// import { getAll, get5First, getRed } from './api/goods';
-// or
-// import * as goodsAPI from './api/goods';
+import { getAll, get5First, getRedGoods } from './api/goods';
 
-export const App: React.FC = () => (
-  <div className="App">
-    <h1>Dynamic list of Goods</h1>
+export const App: React.FC = () => {
+  const [goods, setGoods] = useState<Good[]>([]);
 
-    <button type="button" data-cy="all-button">
-      Load all goods
-    </button>
+  // 'Saw this method when did studen review'
+  // const handleButtonClick = async (callback: Promise<Good[]>) => {
+  //   setGoods(await callback);
+  // };
 
-    <button type="button" data-cy="first-five-button">
-      Load 5 first goods
-    </button>
+  return (
+    <div className="App">
+      <h1 className="title is-1">Dynamic list of Goods</h1>
+      <div className="buttons is-centered">
+        <button
+          type="button"
+          className="button is-info is-rounded is-light"
+          data-cy="all-button"
+          onClick={() => (getAll().then(good => setGoods(good)))}
+        >
+          Load all goods
+        </button>
 
-    <button type="button" data-cy="red-button">
-      Load red goods
-    </button>
+        <button
+          type="button"
+          className="button is-link is-rounded is-light"
+          data-cy="first-five-button"
+          onClick={() => (get5First().then(good => setGoods(good)))}
```

Comment:
In all three `onClick` we basically do the same.
Call some function and use `then` for adding data to the state.
Let's try to create a function that takes a function, call it, and add data to the state